### PR TITLE
streamline interface to 1 command

### PR DESCRIPTION
**Pull request by mholkesvik**
https://app.asana.com/0/795621786541567/889766525163779

### CODE REVIEW

File: samsts/fill.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def fill_scene(
 
     out_tifs = []
     for i, b in enumerate(landsat.BAND_NUMS):
-        out_tif = out_tif_pattern.format(scene=scene, band=b)
+        out_tif = out_tif_pattern.format(date=scene, band=b)
```
Comment (mholkesvik) [inline samsts/fill.py]: confusion here stems from 
https://app.asana.com/0/795621786541567/873625212413998

might be good reason to update now

File: samsts/run.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+# Create formatter and add it to the handler
+formatter = logging.Formatter('%(asctime)s - %(levelname)s: %(message)s')
+handler.setFormatter(formatter)
+
+# Set STDERR handler as the only handler
+logger.handlers = [handler]
+
+
+def run(job, path, row, scenes_to_fill):
```
Comment (mholkesvik) [inline samsts/run.py]: TODO: add click CLI interface?

File: samsts/fill.py (hunk 1 above)
Comment (max-nova) [reply samsts/fill.py]: your call

File: samsts/process.py [hunk 3]
```diff
@@ -37,7 +39,10 @@ def process_full(landsat_fns, scenes_to_fill=None, out_dir=None):
     if scenes_to_fill is None:
         raise NotImplementedError('TODO - automatically pick scenes')
 
-    out_tif_pattern = out_dir.joinpath('{scene}-B{band}.tif')
+    out_tif_pattern = out_dir.joinpath(
+        Path(store_keys.LANDSAT_SAMSTS_RAW).basename()
```
Comment (max-nova) [inline samsts/process.py]: why get so jammed up about using the s3 key... you're going to rename during upload anyways?

File: samsts/run.py (hunk 2 above)
Comment (max-nova) [reply samsts/run.py]: your call.  just update the readme if you do

File: samsts/run.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    # flatten the landsat fns
+    landsat_fns = [f for l in prod_id_to_fns.values() for f in l]
+
+    logging.info("cloud-filling scenes: {}".format(scenes_to_fill))
+    processed_fns = process.process_full(
+        landsat_fns,
+        scenes_to_fill,
+
+        # output directory
+        '/mnt/vol/samsts_processed_p{p}r{r}'.format(p=path, r=row)
```
Comment (max-nova) [inline samsts/run.py]: feels like this should be pulled out to a top-level var

File: samsts/cluster.py [hunk 5]
```diff
@@ -26,14 +26,41 @@ def cluster_stack(stack):
...
     (see find_cluster_neighbors)
     """
-    clusters = kmeans_cluster(stack)
+    logging.info('attempting clustering with default threshold')
+    threshold = CLUSTER_THRESH_INIT
+    clusters = kmeans_cluster(stack, threshold)
+
+    num_clusters = len(clusters)
+
+    while num_clusters < MIN_CLUSTERS or num_clusters > MAX_CLUSTERS:
```
Comment (max-nova) [inline samsts/cluster.py]: or, fancy:

    while not MIN_CLUSTERS <= num_clusters <= MAX_CLUSTERS:

File: samsts/cluster.py [hunk 6]
```diff
@@ -26,14 +26,41 @@ def cluster_stack(stack):
...
+        # iterate on different cluster thresholds until within limits
+
+        if num_clusters < MIN_CLUSTERS:
+            logging.info('Num clusters (%s) below minimum (%s)' % (
+                num_clusters, MIN_CLUSTERS
+            ))
+            clusters = kmeans_cluster(stack, threshold + 0.01)
```
Comment (max-nova) [inline samsts/cluster.py]: Can maybe be more aggressive here.  Change threshold by:

    max(0.1, diff / 50.0)

File: samsts/cluster.py (hunk 5 above)
Comment (mholkesvik) [reply samsts/cluster.py]: cmd-g and suggest!

File: samsts/cluster.py (hunk 6 above)
Comment (mholkesvik) [reply samsts/cluster.py]: no need for 0.1 in this case, will always be diff / 50.0 

1/ 50.0 = 0.02 ?

**Pull request merged**